render/gles2/renderer.c

```diff
@@ -355,10 +356,24 @@ static bool gles2_init_wl_display(struct wlr_renderer *wlr_renderer,
 		struct wl_display *wl_display) {
 	struct wlr_gles2_renderer *renderer =
 		gles2_get_renderer(wlr_renderer);
-	if (!wlr_egl_bind_display(renderer->egl, wl_display)) {
-		wlr_log(WLR_INFO, "failed to bind wl_display to EGL");
-		return false;
+
+	if (renderer->egl->exts.bind_wayland_display_wl) {
```

Comment:
Checking for BINd_WAYLAND_DISPLAY_WL is a separate change or should at least be mentioned in the commit message.

Reply:
The check was performed in `wlr_egl_bind_display` anyway before this commit.

Reply:
And since both are public API, its probably best to leave the check there too.

Reply:
Yes.

I've split the changes into two separate commits.